app/src/main/java/ingsw/group1/findmyphone/contacts/ContactDao.java

```diff
@@ -0,0 +1,28 @@
+package ingsw.group1.findmyphone.contacts;
+
+import androidx.room.Dao;
+import androidx.room.Ignore;
+
+import ingsw.group1.msglibrary.SMSMessage;
+
+/**
+ * Interface extending the BaseDao class for Contact
+ *
+ * @author Giorgia Bortoletti
+ */
+@Dao
+abstract class ContactDao extends BaseDao<Contact> {
+
+    @Ignore
+    public static final String DEFAULT_TABLE_NAME = "contacts_table";
+
+    /**
+     * @return the name of the table containing the {@link Contact} entities.
+     */
+    @Override
+    public String getTableName(){
+        return DEFAULT_TABLE_NAME;
+    }
```

Comment:
I believe count and get all would not work if the name is defined here, the table name should be in the Entity itself, otherwise the class name is used as a default

Reply:
The tests are passed with count and getAll.
In the contact the table name is given from ContactDao as
`@Entity(tableName = ContactDao.DEFAULT_TABLE_NAME)`

How do you suggest to change?

Reply:
I'd just swap them, the constant in the Entity class and importing that constant in the Dao